### ISSUE TITLE
fix(distributions): wrong sigma power in GaussianPCA Woodbury quadratic

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#138]
## Summary

`GaussianPCA._log_prob` divides the Woodbury correction term by `sigma2**2` where it should divide by `sigma2`. `log_prob` is therefore wrong for every `sigma != 1`.

The default is `log_sigma_init = 0.0`, i.e. `sigma^2 = 1` — the single value at which `1/sigma^2 == 1/sigma^4`. That is why the existing tests pass.

## Reproduction

```python
import equinox as eqx, jax, jax.numpy as jnp, jax.random as jr
import jax.scipy.stats as jstats
from gauss_flows import GaussianPCA

jax.config.update("jax_enable_x64", True)

d = GaussianPCA(jr.key(0), event_shape=(6,), latent_dim=2)
d = eqx.tree_at(lambda m: m.log_sigma, d, jnp.log(jnp.sqrt(0.25)))  # sigma^2 = 0.25
x = jr.normal(jr.key(1), (6,))

print(d.log_prob(x))                                                  # +8.43087
print(jstats.multivariate_normal.logpdf(x, d.loc, d.covariance()))    # -6.20853
```

Measured against `scipy.stats.multivariate_normal` with `D=6, K=2`:

| `sigma^2` | `GaussianPCA.log_prob` | Dense reference | Error |
|---|---|---|---|
| 1.0 | −10.71600 | −10.71600 | **0.00000** |
| 0.25 | +8.43087 | −6.20853 | **+14.63940** |
| 4.0 | −10.49525 | −10.48819 | −0.00706 |

Note the `sigma^2 = 0.25` case returns a *positive* log-density for a 6-dimensional continuous distribution — the error is not subtle once `sigma != 1`.

## Root cause

Two Woodbury conventions have been mixed. Writing `Sigma = W W^T + sigma^2 I`:

$$
\Sigma^{-1} = \sigma^{-2} I - \sigma^{-4} W \Psi^{-1} W^\top,
\qquad \Psi = I_K + \sigma^{-2} W^\top W
$$

The `sigma^{-4}` coefficient is correct **only** for `Psi`. The code builds the *other* capacitance matrix,

```python
M = sigma2 * jnp.eye(k) + self.W.T @ self.W      # M = sigma^2 * Psi
```

and since `Psi^{-1} = sigma^2 M^{-1}`, substituting gives

$$
\Sigma^{-1} = \sigma^{-2} I - \sigma^{-2} W M^{-1} W^\top
\quad\Longrightarrow\quad
\mathrm{quad} = \frac{\lVert d \rVert^2 - (W^\top d)^\top M^{-1} (W^\top d)}{\sigma^2}
$$

One factor of `sigma^2` is absorbed by using `M` instead of `Psi`. The code keeps the `Psi`-form exponent while using the `M`-form matrix.

The `log_det` on the following lines is **correct** — `(d - k) * log(sigma2) + log|M|` is the right matrix-determinant-lemma expression for `M`. Only the quadratic is wrong, which is why the error is a pure additive offset in the quadratic and not a scale error.

The inline comment carries the same mistake and must be fixed with the code:

```python
#   quadratic = sigma^-2 * ||diff||^2 - sigma^-4 * (W^T diff)^T M^-1 (W^T diff).
```

## Fix

`src/gauss_flows/_src/distributions/pca.py`, lines 135-136:

```python
        # Sigma^-1 (x - loc) via Woodbury. Note M = sigma^2 * (I + W^T W / sigma^2),
        # so using M rather than the unit-diagonal capacitance absorbs one factor
        # of sigma^2 -- both terms carry sigma^-2, not sigma^-4.
        #   quadratic = sigma^-2 * ( ||diff||^2 - (W^T diff)^T M^-1 (W^T diff) ).
        quad = (jnp.sum(diff**2) - Wt_diff @ y) / sigma2
```

Nothing else in the file changes. `_sigma2`, `covariance`, `_sample`, and the `log_det` computation are all correct.

## Why the tests missed it

`tests/test_distributions_pca.py`:

- `test_log_prob_matches_full_covariance_reference` (line 16) and `test_log_prob_batched` (line 25) both construct `GaussianPCA(...)` with the **default** `log_sigma_init = 0.0`. They parametrize over `latent_dim` but never over `sigma`. At `sigma^2 = 1` the buggy and correct expressions are identical.
- `test_log_prob_stable_for_very_negative_log_sigma` (line 84) does set `log_sigma = -50.0`, but only asserts `jnp.isfinite(lp)` — never correctness.
- `test_sample_moments_match_covariance` exercises `_sample`, which does not touch `_log_prob`.

So the parameter that matters was held fixed at the one value that hides the defect. Worth remembering for the other distributions in this package.

## Acceptance criteria

- [ ] **Parametrize the existing reference test over `sigma`.** Extend `test_log_prob_matches_full_covariance_reference` with `@pytest.mark.parametrize("sigma2", [0.01, 0.25, 1.0, 4.0, 100.0])`, setting `log_sigma` via `eqx.tree_at`, and keep the existing `latent_dim` parametrization. This is the regression test — it must fail on the current code at every `sigma2 != 1.0`.
- [ ] Cover the rank boundary in the same test: `latent_dim` values including `K = 1` and `K = D` (the `(d - k) * log(sigma2)` term vanishes at `K = D`).
- [ ] `test_log_prob_batched` gains a non-unit `sigma`.
- [ ] Upgrade `test_log_prob_stable_for_very_negative_log_sigma` to assert the value against `covariance()`, not just finiteness — the `eps` floor makes `sigma^2 = 1e-12`, where the old code was off by 12 orders of magnitude.
- [ ] Tolerance `1e-8` under `jax_enable_x64`; the existing `atol=1e-5` is fine for the default float32 path.
- [ ] Fix the inline comment as well as the code.
- [ ] Add a `fix:` entry per Conventional Commits (`fix: correct sigma power in GaussianPCA Woodbury quadratic`). Do **not** hand-edit `CHANGELOG.md` — release-please manages it.

## Validation evidence

The proposed fix was applied to a copy of the package and checked against `jax.scipy.stats.multivariate_normal.logpdf(x, loc, covariance())`:

| `sigma^2` | Max error over `(D, K)` in `{(10,2), (6,5), (8,8)}` |
|---|---|
| 1.0 | `0.00e+00` |
| 0.25 | `0.00e+00` |
| 4.0 | `1.78e-15` |
| 0.01 | `2.84e-14` |
| 100.0 | `0.00e+00` |

All combinations agree to `< 1e-8` under x64.

## Downstream note

Anything that trains a `GaussianPCA` base distribution by maximum likelihood has been optimizing a wrong objective — `log_sigma` starts at the one correct point and the gradient is wrong as soon as it moves. `test_as_base_dist_in_gaussianization_flow` should be re-checked after the fix; if it asserts on a fitted value, that value will change.

## Before reporting done

Per `AGENTS.md`:

```bash
uv run pytest -v                        # 0 failures
uv run --group lint ruff check .        # note: '.', not 'src/'
uv run --group lint ruff format --check .
uv run --group typecheck ty check src/gauss_flows
```
